src/Runner.Listener/Runner.cs

```diff
@@ -210,8 +211,14 @@ public async Task<int> ExecuteCommand(CommandSettings command)
                         foreach (var config in jitConfig)
                         {
                             var configFile = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Root), config.Key);
-                            var configContent = Encoding.UTF8.GetString(Convert.FromBase64String(config.Value));
-                            File.WriteAllText(configFile, configContent, Encoding.UTF8);
+                            var configContent = Convert.FromBase64String(config.Value);
+#if OS_WINDOWS
+                            if (configFile == HostContext.GetConfigFile(WellKnownConfigFile.RSACredentials))
+                            {
+                                configContent = ProtectedData.Protect(configContent, null, DataProtectionScope.LocalMachine);
+                            }
+#endif
+                            File.WriteAllBytes(configFile, configContent);
                             File.SetAttributes(configFile, File.GetAttributes(configFile) | FileAttributes.Hidden);
                             Trace.Info($"Save {configContent.Length} chars to '{configFile}'.");
```

Comment:
`configContent.Length` might be off since configContent become bytes now.

Reply:
Good point! Is it enough to update the log message?

```suggestion
                            Trace.Info($"Saved {configContent.Length} bytes to '{configFile}'.");
```

Reply:
looks good. 👍